src/averageAges.js

```diff
@@ -55,7 +67,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMother = people.filter(child =>
+    child.mother && people.some(parent => parent.name === child.mother)
+  );
+  const filteredChildrenWithMother = onlyWithSon
+    ? childrenWithMother.filter(child => child.sex === 'm')
+    : childrenWithMother;
+  const ageDifferences = filteredChildrenWithMother.map(person =>
+    person.born - people.find(parent => parent.name === person.mother).born
+  );
+
+  return ageDifferences.reduce((sum, ageDiff) =>
+    sum + ageDiff, 0
+  ) / ageDifferences.length;
```

Comment:
If you used const totalAge in the last functions, why you skip that here? Using one solution kind for all similar functions it's a better practice.